### PR TITLE
MarkLogic ExecutePlan support and regression tests

### PR DESCRIPTION
**Pull request by wemrysi**
Adds an implementation of `ExecutePlan` for MarkLogic and enables a handful of regression tests.

Some of the regression tests that were using the implicit numerical field names were modified to use explicit aliases to avoid a mismatch in expectations due to the current handling of numeric field names, which seemed reasonable as the field names themselves aren't what is being tested.

### CODE REVIEW

File: marklogic/src/main/scala/quasar/physical/marklogic/qscript/MapFuncPlanner.scala [hunk 1]
```diff
@@ -83,25 +83,25 @@ object MapFuncPlanner {
 
     // math
     case Negate(x)      => (-x).point[F]
-    case Add(x, y)      => (x + y).point[F]
-    case Multiply(x, y) => (x * y).point[F]
-    case Subtract(x, y) => (x - y).point[F]
-    case Divide(x, y)   => (x div y).point[F]
-    case Modulo(x, y)   => (x mod y).point[F]
+    case Add(x, y)      => binOp[F](x, y)(_ + _)
```
Comment (mossprescott) [inline marklogic/src/main/scala/quasar/physical/marklogic/qscript/MapFuncPlanner.scala]: Is this about evaluation order? Cleaning up the xquery syntax? Again, mostly curious and not a blocker.

**Comment by mossprescott** [comment]
Looks good to me and awesome to see a dozen or so tests passing out of the gate.

Unfortunately, it now needs a merge from master. 😞

File: marklogic/src/main/scala/quasar/physical/marklogic/qscript/MapFuncPlanner.scala (hunk 1 above)
Comment (wemrysi) [inline marklogic/src/main/scala/quasar/physical/marklogic/qscript/MapFuncPlanner.scala]: Yes, as `x` and `y` can be arbitrary expressions, there was some ambiguity in certain situations. However, it is a bit more hammer than is needed in many situations (like constants or other basic expressions), that can be something dealt with later as this change should only improve correctness possibly at the expense of clarity.

**Comment by wemrysi** [comment]
@mossprescott Ok, conflicts resolved and building. Unfortunately, I had to disable three of the tests that were passing prior to the recent commits to master as they now fail :disappointed: The ones involving joins fail with a "non theta join condition found" error (first I've heard of this one), not sure why the third is failing.

**Comment by sellout** [comment]
@wemrysi Hrmm, is that a regression _since_ my join fixes the other day? (It’s hard to tell from the PR what was new in the merge.)

Also, why “skip” rather than “pending”?

**Comment by wemrysi** [comment]
@sellout Yes, the passing build above was late in the day yesterday CDT and merging in master just now caused the `projectIndexWithObjectFlatten`, `3WayJoin` and `innerEquiJoin`tests to fail.

**Comment by wemrysi** [comment]
And "skip" because running all of them as "pending" locally just takes too long at the moment.

**Comment by wemrysi** [comment]
@sellout Sorry, read too quickly, no it looks like both the success and the failure included your join fixes, the interesting `QScript` related commits between the success and the failure (determined by limiting to your and @alissapajer's commits and filtering out merges, changelog additions) are
```
ab3c5d30b389e6e55384cf2af403b558a9f92b97 Implement Distinct & fix Union.
a48370f61ad66d8ef354f77af30a465db270bcb9 Clean up Normalizable.
ca836b2b418dd1657ce6738b851b65f7afca0b7c A collection of QScript fixes.
7cebbd640c31008202cf492d25a040dec829fc36 Refactor Coalesce and Normalizable.
```

**Comment by sellout** [comment]
Ah! The join tests hit #1539.

I’m not sure why the third is failing – it at least generates QScript. At a quick glance it looks reasonable (definitely would take some time to dig through the provenance details) – it has a Join in it, but it should, since it has to join the collection with a flattened version of the collection (which is not the same as the self-joins in #1539).

**Comment by mossprescott** [comment]
Going to assume these issues with specific queries can be worked out in the future and merging this as is.

**Pull request merged**